banzai/image/image.py

```diff
@@ -18,60 +18,42 @@
 
 class Image(object):
 
-    def __init__(self, pipeline_context, filename=None, data=None, header=None,
-                 extension_headers=None, bpm=None):
-        if header is None:
-            header = {}
-
+    def __init__(self, pipeline_context, filename: str = None, data=None, header: Header = None,
```

Comment:
When an `Image` object is created, where does the `header` come from? Is it not contained in the file `filename` that is being passed in here?

Reply:
This could be factory class methods.